machFramework/contribute.rst

```diff
@@ -0,0 +1,130 @@
+The version we use can be installed with:
+
+.. prompt:: bash
+
+    pip install black==22.3.0
+
+``black`` can then be run at the project root with:
+
+.. prompt:: bash
+
+    black . -l 120
+
+This will automatically format all Python files.
+
+We use `flake8 <https://flake8.pycqa.org/en/latest/>`_ for linting in Python.
+The recommended version and any necessary dependencies are in `this file <https://github.com/mdolab/.github/blob/main/flake8-requirements.txt>`_.
+You can install them by calling ``pip install`` for each individually or copying the contents of that file into one on your machine and typing:
+
+.. prompt:: bash
+
+    pip install flake8-requirements.txt
+
+The configuration file we use for ``flake8`` is a combination of `this .flake8 file <https://github.com/mdolab/.github/blob/main/.flake8>`__ and the one at the root of the respective repository.
+``flake8`` can then be run at the project root with:
+
+.. prompt:: bash
+
+    flake8 .
+
+If there are any PEP-8 violations, ``flake8`` will print out the nature of the violation.
+
+Fortran
+^^^^^^^
+
+We use `fprettify <https://github.com/pseewald/fprettify>`_ for formatting Fortran codes. 
+The version we use can be installed with:
+
+.. prompt:: bash 
+
+    pip install fprettify==0.3.7
+
+The configuration file for ``fprettify`` is at the root of the respective repository.
+If there isn't a repo-specific config, `this global fprettify config <https://github.com/mdolab/.github/blob/main/.fprettify.rc>`_ is used.
+``fprettify`` can then be run at the project root using `this fprettify bash script <https://github.com/mdolab/.github/blob/main/azure/fprettify.sh>`_ with:
+
+.. prompt:: bash
+
+    ./fprettify.sh
+
+C/C++
+^^^^^
+
+We use `clang-format <https://clang.llvm.org/>`_ to format C/C++ codes. 
+Please install **version 10** following its documentation.
+
+The configuration file for ``clang-format`` is at the root of the respective repository.
+If there isn't a repo-specific config, `this global clang-format config <https://github.com/mdolab/.github/blob/main/.clang-format>`_ is used.
+``clang-format`` can then be run at the project root using `this clang-format bash script <https://github.com/mdolab/.github/blob/main/azure/clang-format.sh>`_ with:
+
+.. prompt:: bash
+
+    ./clang-format.sh
+
+
+.. warning::
+    For a PR to be accepted it must pass formatting checks with the relevant formatter and/or linter.
+
+Documentation
+-------------
+When you add or modify code, make sure to provide relevant documentation that explains the new code.
+This should be done in code via docstrings and comments as well as in the Sphinx documentation if you add a new feature or capability.
+Look at the ``.rst`` files in the ``doc`` section of each repo.
+
+Building the documentation requires ``sphinx`` and ``numpydoc``, as well as the Sphinx RTD theme.
+To install these dependencies, type:
+
+.. prompt:: bash
+
+    pip install sphinx numpydoc sphinx-rtd-theme sphinx-mdolab-theme
```

Comment:
What I meant is that this should just be `pip install sphinx-mdolab-theme`. This will also install all the prerequisites. The text above should also be changed to match this.